### PR TITLE
MAINT: Misc numeric cleanup

### PR DESCRIPTION
**Pull request by eric-wieser**
A bunch of small and somewhat orthogonal cleanups, all helping a little with #10151, and maybe #6103

### CODE REVIEW

File: numpy/core/numerictypes.py [hunk 1]
```diff
@@ -283,29 +296,33 @@ def bitname(obj):
...
+        allTypes[name] = cls
 _add_types()
 
+# This is the priority order used to assign the bit-sized NPY_INTxx names, which
+# must match the order in npy_common.h in order for NPY_INTxx and np.intxx to be
+# consistent.
+# If two C types have the same size, then the earliest one in this list is used
+# as the sized name.
+_int_ctypes = ['long', 'longlong', 'int', 'short', 'byte']
+_uint_ctypes = list('u' + t for t in _int_ctypes)
```
Comment (mhvk) [inline numpy/core/numerictypes.py]: Total nitpick, but just `['u' + t for t in _int_ctypes]` is faster (and clearer)

**Review by mhvk — COMMENTED** [review]
Even just the added comments are worth it! Looks good modulo a small nitpick.

File: numpy/core/numerictypes.py (hunk 1 above)
Comment (eric-wieser) [reply numpy/core/numerictypes.py]: A list comprehension would leak `t` into the module scope, unlike the above which does not.

Comment (mhvk) [reply numpy/core/numerictypes.py]: Ah, had not thought about that - OK, would seem all good to go then!

**Comment by eric-wieser** [comment]
Follow-up at #10151

**Pull request merged**